### DISCUSSION TITLE
Clipboard access from Sandbox

### DISCUSSION THREAD

**Discussion by Kethen** [hecrj/iced#1832] [Q&A] [answered]
Hello, I'm new to iced, new to rust in general.

I tried to write some text to clipboard using the `iced::clipboard` module, but upon closer inspection it seems to be more for internal use. The `iced_native::clipboard` module meanwhile contains a `Null` implementation only.

I'm guessing the only way to have clipboard access is through `iced_winit::clipboard::Clipboard` with a `winit::window::Window` handle, how can one do that while using `Sandbox`? Or, how can one access the clipboard besides through the `TextInput` widget with `Sandbox`?

Thanks!

**Comment by hecrj** [comment, marked as answer]
You need to upgrade to the `Application` trait and use [the `clipboard` commands](https://docs.rs/iced/latest/iced/clipboard/index.html).
